lib/api/src/index.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { Component, useContext } from 'react';
+import memoize from 'memoizerific';
+
+import Events from '@storybook/core-events';
+import { RenderData as RouterData } from '@storybook/router';
+import { Collection, Types } from '@storybook/addons';
+import initProviderApi from './init-provider-api';
+
+import { createContext } from './context';
+import Store from './store';
+import getInitialState from './initial-state';
+
+import initAddons, { SubAPI as AddonsAPI } from './modules/addons';
+import initChannel, { SubAPI as ChannelAPI } from './modules/channel';
+import initNotifications, {
+  SubState as NotificationState,
+  SubAPI as NotificationAPI,
+} from './modules/notifications';
+import initStories, { SubState as StoriesSubState, SubAPI as StoriesAPI } from './modules/stories';
+import initLayout, { SubState as LayoutSubState, SubAPI as LayoutAPI } from './modules/layout';
+import initShortcuts, {
+  SubState as ShortcutsSubState,
+  SubAPI as ShortcutsAPI,
+} from './modules/shortcuts';
+import initURL, { QueryParams } from './modules/url';
+import initVersions, { SubState as VersionsSubState } from './modules/versions';
+
+const ManagerContext = createContext({ api: undefined, state: getInitialState({}) });
+
+const { STORY_CHANGED, SET_STORIES, SELECT_STORY } = Events;
+
+export type Module = StoreData & RouterData & ProviderData;
+
+export type State = Other &
+  LayoutSubState &
+  StoriesSubState &
+  NotificationState &
+  VersionsSubState &
+  RouterData &
+  ShortcutsSubState;
+
+export type API = AddonsAPI &
+  ChannelAPI &
+  StoriesAPI &
+  LayoutAPI &
+  NotificationAPI &
+  ShortcutsAPI &
+  OtherAPI;
+
+interface OtherAPI {
+  [key: string]: any;
+}
+interface Other {
+  customQueryParams: QueryParams;
+
+  notifications: Notification[];
+
+  [key: string]: any;
+}
+
+export interface Combo {
+  api: API;
+  state: State;
+}
+
+interface Provider {
+  handleAPI(api: API): void;
+  renderPreview(): void;
+  getElements(type: Types): Collection;
+  [key: string]: any;
```

Comment:
Can this `any` be solved by a generic and does it make sense?
`interface Provider<T> { [key:string]: T; }`